File: playbook-java/src/main/java/io/infinitic/playbook/java/asyncMethod/services/RemoteServiceImpl.java
package io.infinitic.playbook.java.asyncMethod.services;

import io.infinitic.playbook.java.Worker;
import io.infinitic.tasks.Task;
import java.time.LocalDateTime;

public class RemoteServiceImpl implements RemoteService {
    @Override
    public void run(String id, long input)  {
        log("start processing " + id);
        try {
            Thread.sleep(input*1000);
        } catch (InterruptedException e) {
            e.printStackTrace();
        }
        log("stop processing " + id);

    }

    private void log(String msg) {
        System.out.println(
                LocalDateTime.now().format(Worker.formatter) + " - Service  " + Task.getTaskId() + " - " + msg
        );
    }
}

File: playbook-kotlin/src/main/kotlin/io/infinitic/playbook/kotlin/asyncMethod/services/RemoteServiceImpl.kt
package io.infinitic.playbook.kotlin.asyncMethod.services

import io.infinitic.playbook.kotlin.formatter
import io.infinitic.tasks.Task
import java.time.LocalDateTime

class RemoteServiceImpl: RemoteService {
    override fun run(id: String, input: Long) {
        log("start processing $id")
        Thread.sleep(input*1000)
        log("stop processing $id")
    }

    private fun log(msg: String) =
        println("${LocalDateTime.now().format(formatter)} - Service  ${Task.taskId} - $msg")
}